### PR TITLE
Add Batch Management SDK for node.js.

### PR DESCRIPTION
**Pull request by xingwu1**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@xingwu1__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (xingwu). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: test/services/batchManagement/batchManagementClient-tests.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    suite.setupTest(done);
+  });
+  
+  afterEach(function (done) {
+    suite.baseTeardownTest(done);
+  });
+
+  describe('operations', function () {
+        
+        it('should get subscription quota successfully', function (done) {
```
Comment (amarzavery) [inline test/services/batchManagement/batchManagementClient-tests.js]: indentation 2 spaces

File: lib/services/batchManagement/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "azure-arm-batch",
+  "author": "Microsoft Corporation",
+  "contributors": [
+    "Wu, Xing <xingwu@microsoft.com>"
+  ],
+  "version": "0.15.0",
```
Comment (amarzavery) [inline lib/services/batchManagement/package.json]: This is your first version.
https://www.npmjs.com/search?q=azure-arm-batch
It should be "0.1.0"

File: lib/services/batchManagement/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "node": ">= 0.10.0"
+  },
+  "licenses": [
+    {
+      "type": "Apache 2.0",
+      "url": "http://www.apache.org/licenses/LICENSE-2.0"
+    }
+  ],
+  "dependencies": {
+    "ms-rest": "^1.8.0",
```
Comment (amarzavery) [inline lib/services/batchManagement/package.json]: ^1.10.0 for both ms-rest and ms-rest-azure

**Comment by amarzavery** [comment]
please pull from upstream master and regenerate using the latest version of codegen.

**Comment by xingwu1** [comment]
Update the code based on feedback.

**Comment by xingwu1** [comment]
@amarzavery  All the issues you mentioned had already fixed in the update. Could you review the update?

**Pull request merged**